src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerArgumentBinder.cs

```diff
@@ -111,10 +132,37 @@ public class ControllerArgumentBinder : IControllerActionArgumentBinder
                 throw new ArgumentNullException(nameof(controllerContext));
             }
 
+            var valueProvider = await CompositeValueProvider.CreateAsync(
+                controllerContext,
+                controllerContext.ValueProviderFactories);
+
+            return await BindModelAsync(parameter, controllerContext, valueProvider);
+        }
+
+        public async Task<ModelBindingResult?> BindModelAsync(
+            ParameterDescriptor parameter,
+            ControllerContext controllerContext,
+            IValueProvider valueProvider)
```

Comment:
The argument binder will look for the prefix no matter what. We could try to change that, but that's another change.

Lazy property doesn't work because creating VPs is async.